### PR TITLE
Fix error from launching violin plot in scRNAseq gene expression (GDC…

### PR DESCRIPTION
**Pull request by congyu-lu**
… and native)

## Checklist

[Check each task](https://github.com/stjude/proteinpaint/wiki/Pull-Request-Checklist) that has been performed or verified to be not applicable.
- [x] Tests: added and/or passed unit and integration tests, or N/A
- [x] Todos: commented or documented, or N/A
- [x] Notable Changes: updated release.txt, prefixed a commit message with "fix:" or "feat:", added to an internal tracking document, or N/A

### CODE REVIEW

**Review by airenzp — APPROVED** [review]
Thanks for the fix, I noticed this bug the other day but did not had time to look at it. Great that you solve it.

File: server/src/termdb.matrix.js [hunk 1]
```diff
@@ -233,7 +233,7 @@ async function getSampleData(q, ds, onlyChildren = false) {
 		} else if (tw.term.type == TermTypes.SINGLECELL_CELLTYPE) {
 			const data = await q.ds.queries?.singleCell?.data.get({ sample: tw.term.sample, plots: [tw.term.plot] })
```
Comment (xzhou82) [inline server/src/termdb.matrix.js:234]: not handling when data is undefined. i'm improving it

**Comment by xzhou82** [comment]
adding error handling to ky request

File: server/routes/termdb.singlecellSamples.ts [hunk 2]
```diff
@@ -307,7 +306,13 @@ function gdc_validateGeneExpression(G, ds, genome) {
 			}
 
 			const { host } = ds.getHostHeaders(q)
-			const out = await ky.post(joinUrl(host.rest, 'scrna_seq/gene_expression'), { timeout: false, json: body }).json()
+
+			const t = Date.now()
+			//const out = await ky.post(joinUrl(host.rest, 'scrna_seq/gene_expression'), { timeout: false, json: body }).json()
+			const response = await ky.post(joinUrl(host.rest, 'scrna_seq/gene_expression'), { timeout: false, json: body })
```
Comment (xzhou82) [inline server/routes/termdb.singlecellSamples.ts:312]: @siosonel please take a look if this err handling is okay

**Review by xzhou82 — APPROVED** [review]

**Pull request merged**